### DISCUSSION TITLE
Inertia JS Link component and in-app browser misbehavior

### DISCUSSION THREAD

**Discussion by igorkang-smc** [inertiajs/inertia#1914] [Help]
### Version: 1.0.11

- `@inertiajs/react` version: 1.0.11

### Describe the problem:

1. Open the search bar in the Naver app (iOS), enter 'inertia js,' and navigate to home page.
2. Navigate to several different pages on the home page and click back button.
Clicking the back button will exit the site and return directly to the search results, as if previously visited pages were opened as modals.

I guess it is problem of Naver app more than Inertia itself, but I want to find out possible reasons of such behavior!

**Comment by RobertBoes** [comment]
When navigating with Inertia it pushes to the History API (https://developer.mozilla.org/en-US/docs/Web/API/History/pushState), so the back button in a browser would just work as intended, you can see that behavior in a regular browser. So my guess would be, the button in that specific app wouldn't navigate back, but rather just exit instead of navigating back.

**Reply by igorkang-smc** [reply]
The problem is that, unlike other sites where the back button works as expected, this issue occurs only on sites which made by inertiajs. I have checked several sites built with frameworks like React and Vue, and back button function correctly.

I tested multiple sites built with Inertiajs, such as laracasts.com, inertiajs.com, codecourse.com. On all these sites, the back button ignores all visited pages and just exits back to the search results page, as if only the main page was visited!

On my site, I replaced some Link components with simple a href tags. As a result, the app started recognizing page visits made through the a tag and back button navigates to previous pages. 

Any thoughts about reason of such behavior? 

By the way, Naver App is the most popular search app in South Korea, and people use it more frequently than Google, I guess.

**Reply by RobertBoes** [reply]
> I tested multiple sites built with Inertiajs, such as laracasts.com, inertiajs.com, codecourse.com. On all these sites, the back button ignores all visited pages and just exits back to the search results page, as if only the main page was visited!

Well, all of these are built with Inertia, but that doesn't immediately mean Inertia is the issue. As I said, Inertia navigates using `window.history.pushState()`, which is a standard way to perform navigation in an SPA. There is no other way to navigate within an SPA. So that just sounds like the app you're talking about doesn't work with SPAs and/or does not implement the HIstory API properly.

**Comment by corentinclichy** [comment]
Hi I do have the same exact issue. Nevertheless this problem occured when I am using a mobile website

**Reply by riotinri** [reply]
Me too, but only on chrome and ios in combination. android working, desktop working and safari on ios working.

**Reply by AbdullahAgsar** [reply]
@corentinclichy @riotinri I have the same problem. Did you find any solution?

**Reply by corentinclichy** [reply]
Seems that the fix have been fix 🔥
Available in the beta release : 
https://github.com/inertiajs/inertia/issues/1954#issuecomment-2427471572

**Comment by cahuayamarcial** [comment]
Tengo el mismo problema ¿Alguien que pudo solucionar ese problema?
Pasa con Chrome + iOS.

**Reply by AbdullahAgsar** [reply]
@cahuayamarcial  Unfortunately, I couldn’t find a permanent solution. I had to disable the SPA (Single Page Application) only on iOS Chrome and Edge browsers. Instead of handling navigation through the router, I used regular <a> tags for page transitions.

**Reply by BitshapeSolutions** [reply]
Abdullah, how did you disable the SPA functionality on iOS Chrome and Edge browsers only? I'm having the same issue with a Laravel + Inertia + Vue app.
